src/prof_sys.c

```diff
@@ -544,16 +545,23 @@ prof_dump_filename(tsd_t *tsd, char *filename, char v, uint64_t vseq) {
 	assert(tsd_reentrancy_level_get(tsd) == 0);
 	const char *prof_prefix = prof_dump_prefix_get(tsd_tsdn(tsd));
 
-	if (vseq != VSEQ_INVALID) {
-	        /* "<prefix>.<pid>.<seq>.v<vseq>.heap" */
-		malloc_snprintf(filename, DUMP_FILENAME_BUFSIZE,
-		    "%s.%d.%"FMTu64".%c%"FMTu64".heap",
-		    prof_prefix, prof_getpid(), prof_dump_seq, v, vseq);
-	} else {
-	        /* "<prefix>.<pid>.<seq>.<v>.heap" */
-		malloc_snprintf(filename, DUMP_FILENAME_BUFSIZE,
-		    "%s.%d.%"FMTu64".%c.heap",
-		    prof_prefix, prof_getpid(), prof_dump_seq, v);
+	if (!opt_prof_dump_overwrite) {
+	    if (vseq != VSEQ_INVALID) {
+	            /* "<prefix>.<pid>.<seq>.v<vseq>.heap" */
+		    malloc_snprintf(filename, DUMP_FILENAME_BUFSIZE,
+		        "%s.%d.%"FMTu64".%c%"FMTu64".heap",
+		        prof_prefix, prof_getpid(), prof_dump_seq, v, vseq);
+	    } else {
+	            /* "<prefix>.<pid>.<seq>.<v>.heap" */
+		    malloc_snprintf(filename, DUMP_FILENAME_BUFSIZE,
+		        "%s.%d.%"FMTu64".%c.heap",
+		        prof_prefix, prof_getpid(), prof_dump_seq, v);
+	    }
+	} else { // opt_prof_dump_overwrite
+	    /* "<prefix>.<pid>.<v>.heap" */
+	    malloc_snprintf(filename, DUMP_FILENAME_BUFSIZE,
+	        "%s.%d.%c.heap",
+	        prof_prefix, prof_getpid(), v);
```

Comment:
Is the new mode intended for every type of dump, or just the interval triggered dump? My guess is interval triggered dump only? If so you might not want to make things different for other types of dumps (e.g. manual dumps).

Reply:
I was planning on having the mode apply to all modes' autogenerated filenames, for consistency's sake and code understandability. Definitely don't feel strongly about this though - my usecase will need to be able to parse both the 'overwrite' filenames and 'normal'. 

Happy to add a check for `v == 'i'` in this code and comment explaining why.

Reply:
By all modes you are basically meaning that there will ever be one dump file of each mode on the file system? It might make sense. I'll leave the decision to you then. The particular mode probably worth more thinking is the peak triggered dumping, a.k.a. `gdump`.